src/rules/unnecessaryConstructorRule.ts

```diff
@@ -39,12 +57,49 @@ export class Rule extends Lint.Rules.AbstractRule {
     public static FAILURE_STRING = "Remove unnecessary empty constructor.";
 
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        return this.applyWithFunction(sourceFile, walk);
+        const options: Options = {
+            unnecessarySuperCall:
+                this.ruleArguments.length !== 0 &&
+                (this.ruleArguments[0] as { "unnecessary-super-call"?: boolean })[
+                    "unnecessary-super-call"
+                ] === true,
+        };
+
+        return this.applyWithFunction(sourceFile, walk, options);
     }
 }
 
-const isEmptyConstructor = (node: ts.ConstructorDeclaration): boolean =>
-    node.body !== undefined && node.body.statements.length === 0;
+const containsStatements = (statements: ts.NodeArray<ts.Statement>) => statements.length > 0;
+
+const containsSuper = (statements: ts.NodeArray<ts.Statement>): boolean => {
+    for (const statement of statements) {
+        if (
+            isExpressionStatement(statement) &&
+            isCallExpression(statement.expression) &&
+            ts.SyntaxKind.SuperKeyword === statement.expression.expression.kind
+        ) {
+            return true;
+        }
+    }
+
+    return false;
+};
+
+const isEmptyOrContainsOnlySuper = (node: ts.ConstructorDeclaration, options: Options): boolean => {
+    if (node.body) {
+        const { unnecessarySuperCall } = options;
+
+        if (!containsStatements(node.body.statements)) {
```

Comment:
Seems like it'd be a bit less code to just say `if (node.body.statements.length === 0)`, no? Let's change to that.

Reply:
Done